src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java

```diff
@@ -104,13 +105,115 @@ String getLogStreamName(String thingName) {
+                        .log("Error reading file contents. Invalid start position");
+                return attempt;
+            }
+
+            StringBuilder data = new StringBuilder(line);
+
+            // Run the loop until we detect that the log file is completely read, or that we have reached the max
+            // message size or if we detect any IOException while reading from the file.
+            while (!reachedMaxSize) {
+                long tempStartPosition = reader.position();
+                String partialLogLine = reader.readLine();
+
+                // If we do not get any data from the file, we have reached the end of the file.
+                // and we add the log line into our input logs event list.
+                if (partialLogLine == null) {
+                    processLogLine(totalBytesRead,
+                            cpInfo.getDesiredLogLevel(), logStreamName,
+                            logStreamsMap, data, fileInfo.getFileHash(), startPosition,
+                            cpInfo.getName(),
+                            tempStartPosition, logFile.lastModified(), logFileGroup);
+                    break;
+                }
+
+                // If the new log line read from the file matches the start pattern, that means
+                // the string builder we have appended data to until now, has a complete log line.
+                // Let's add that in the input logs event list.
+                // The default pattern is checking if the line starts with non-white space
+                if (checkLogStartPattern(cpInfo, partialLogLine)) {
+                    reachedMaxSize = processLogLine(totalBytesRead,
+                            cpInfo.getDesiredLogLevel(), logStreamName,
+                            logStreamsMap, data, fileInfo.getFileHash(), startPosition,
+                            cpInfo.getName(),
+                            tempStartPosition, logFile.lastModified(), logFileGroup);
+                    data = new StringBuilder();
+                }
+
+                // Need to read more lines until we get a complete log line. Let's add this to the SB.
+                data.append(partialLogLine);
+            }
+        } catch (IOException e) {
+            // File probably does not exist.
+            logger.atError().cause(e).log("Unable to read file {}", logFile.getAbsolutePath());
+        }
+
+        return attempt;
+    }
+
     /**
      * Gets CW input log events from the component which processLogFiles need to be uploaded to CloudWatch.
      *
      * @param componentLogFileInformation log files information for a component to read logs from.
      * @return CloudWatch attempt containing information needed to upload logs from the component to the cloud.
      */
-    public CloudWatchAttempt processLogFiles(ComponentLogFileInformation componentLogFileInformation) {
+    public CloudWatchAttempt processLogFilesDeprecated(ComponentLogFileInformation componentLogFileInformation) {
```

Comment:
Tagged this as deprecated for now we need to go and update the tests that are directly calling this method

Reply:
We will delete this in the next PR

Reply:
the new function shouldn't have this name, so do we need to rename it? it just makes the PR bigger for no reason

Reply:
Shall we directly delete it now?

Reply:
@jbutler not sure I get your comment. Are you saying keep the old method being called `processLogFiles` and have the new one have a different method? I think the effect on the size of the PR is the same because we have to change the tests that directly called the old one to call the new one

Reply:
The new method isn't processing files. It's processing a single file. So there was no reason to rename this is what I was saying

Reply:
Got it, it was being called directly on test though the only real saving is not seeing that line change so from this what we can do is revert the naming of this and change the name of the one above to something like `processLogFile` or `addLogFileToAttempt`?